### ISSUE TITLE
Consume with end

### ISSUE THREAD

**Issue by sehz** [infinyon/fluvio#1940]
In order to fulfill use case for [bulk read and write](https://github.com/infinyon/fluvio/issues/1898), we need to limit consume when it reaches certain criteria instead of waiting for further inputs.  In order to that we should provide 'end` options:

```
$ fluvio consume my_topic -B --end <offset>
```

<end_option> is optional.  if it is not specified, it would read to current offset at time of reading.

**Comment by sehz** [comment]
@bohlmannc, interested in this issue?

**Comment by bohlmannc** [comment]
Yes, I will look into it.

**Comment by tjtelan** [comment]
We have the `disable_continuous` option which sort of implements this by closing the stream when we've reached the end of a topic. We would need to pass in an optional ending offset to the stream handling loop so we can exit early

https://github.com/infinyon/fluvio/blob/e9807a82eb9fdffc7391a7f39cfa8f2a75d6ccb7/crates/fluvio-cli/src/consume/mod.rs#L353-L358

**Comment by bohlmannc** [comment]
Hey tj just to clarify, when reading records from a stream, the offset is not returned as part of the result, so would it suffice to just count how many records are successfully read from the stream and then exit early when that count equals the optional ending offset? I think I was overcomplicating things before I saw your comment above.

**Comment by tjtelan** [comment]
@bohlmannc The [`Record`](https://docs.rs/fluvio/0.10.0/fluvio/consumer/struct.Record.html) type we receive from `stream_with_config` (which we call within `consumer_records_stream`) does include the offset returned alongside the data content.

We should add an optional command-line flag (to `ConsumeOpt`) to take in this end offset, and use it when building the consumer's config. I was thinking this end offset would be added to `ConsumerConfig` as an `Option<i64>` (to match the numerical type we use inside `Record`). Then we can compare offsets in the stream fetch loop after printing the record.

https://github.com/infinyon/fluvio/blob/8e77bda1b0b7315c2c038965ebb3ba25fcc57b1e/crates/fluvio/src/consumer.rs#L711-L727

After we close the stream, we should print a message stating that we've exited the consumer stream bc we've reached the offset threshold.
